### PR TITLE
refactor: remove deprecations and update them to current API

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

_Provide description of this PR and changes, if linked Jira ticket doesn't cover it in full._

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/services/SnykCliAuthenticationService.kt [hunk 1]
```diff
@@ -47,7 +48,7 @@ class SnykCliAuthenticationService(val project: Project) {
         downloadCliIfNeeded()
         if (getCliFile().exists()) executeAuthCommand()
         if (isAuthenticated) executeGetConfigApiCommand()
-
+        LanguageServerWrapper.getInstance().updateConfiguration()
```
Comment (teodora-sandu) [inline src/main/kotlin/io/snyk/plugin/services/SnykCliAuthenticationService.kt:51]: Question: Was this something we should have added before and you just noticed?

Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykCliAuthenticationService.kt:51]: It was a merge problem. The method was renamed...

Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykCliAuthenticationService.kt:51]: Ok, let me rephrase. I noticed, that the token was not updated in Language Server after authentication. So I added it.

**Review by teodora-sandu — APPROVED** [review]
I've tested the Settings panel (changing the ignore settings and severities) and also had a look at Code Security and Code Quality issues. I haven't been able to test the other types of issues because of the bug we have with disabling/enabling other types of scanning from the Settings panel but I think it should be okay
